File: atcoder_selected_100/ruby/abc122_b.rb
# 3
# ABC122/B
# 全探索:全列挙
# 尺取法

require "set"

ACGT = "ACGT".chars.to_set
S = gets.chomp.chars

l = r = 0
ans = []
while (l < S.size)
  while r < S.size && ACGT.include?(S[r])
    r += 1
  end
  ans << r - l
  r += 1
  l = r
end

puts ans.max

File: atcoder_selected_100/kotlin/abc122_b.kt
/**
 * 3
 * ABC122/B
 * 全探索:全列挙
 * 尺取法
 */

 fun main() {
  val ACGT = "ACGT".toSet()
  val S = readLine()!!.toList()
  var l = 0
  var r = 0
  var ans = 0
  while (l < S.size) {
    while (r < S.size && (S[r] in ACGT)) { r++ }
    ans = maxOf(ans, r - l)
    l = ++r
  }
  println(ans)
}
